### PR TITLE
[requirements] Adjust ML requirements to work with Mac ARM (M1/M2)

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Some of our dependencies are incompatible with MacOS ARM architectures. This PR updates the requirements.txt files so that dependency installation works out of the box.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements/ml/requirements_tune.txt [hunk 1]
```diff
@@ -9,18 +9,21 @@ ConfigSpace==0.4.18
 dragonfly-opt==0.1.6
 flaml==1.1.1
 freezegun==1.1.0
-gluoncv==0.10.1.post0
+# Requires decord which is unavailable for arm64
+gluoncv==0.10.1.post0; platform_machine != "arm64"
```
Comment (matthewdeng) [inline python/requirements/ml/requirements_tune.txt:13]: Generally looks good to me, anything we need to do to test this change?

**Review by matthewdeng — APPROVED** [review]

File: python/requirements/ml/requirements_tune.txt (hunk 1 above)
Comment (krfricke) [reply python/requirements/ml/requirements_tune.txt:13]: It does not affect our CI machines and docker images (or the parts that do are captured in the respective build processes), so once CI passes we should be fine to go. The current state for mac arm machines was that install didn't work at all, so this PR should be a strict improvement.

File: python/requirements/ml/requirements_dl.txt [hunk 2]
```diff
@@ -11,7 +12,7 @@ tensorflow-probability==0.19.0
 torch==1.13.0
 torchvision==0.14.0
 torch-scatter==2.1.0
-torch-sparse==0.6.15
+torch-sparse==0.6.16
```
Comment (ddelange) [inline python/requirements/ml/requirements_dl.txt:15]: for correctness also needs updating here: https://github.com/ray-project/ray/blob/ray-2.3.0/python/requirements/ml/requirements_ml_docker.txt#L14

File: python/requirements/ml/requirements_dl.txt [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # These requirements are used for the CI and CPU-only Docker images so we install CPU only versions of torch.
 # For GPU Docker images, you should install requirements_ml_docker.txt afterwards.
 
-tensorflow==2.11.0
+tensorflow==2.11.0; sys_platform != 'darwin' or platform_machine != 'arm64'
+tensorflow-macos==2.11.0; sys_platform == 'darwin' and platform_machine == 'arm64'
```
Comment (ddelange) [inline python/requirements/ml/requirements_dl.txt:5]: on recent macs you'll probably also want gpu acceleration using `tensorflow-metal` ref https://developer.apple.com/metal/tensorflow-plugin/

**Pull request merged**